### PR TITLE
[NVTX] remove pragma hide

### PR DESCRIPTION
**Pull request by simonbyrne**
Thanks again to @giordano: https://github.com/JuliaPackaging/Yggdrasil/pull/5903#issuecomment-1322760932

I also added `-ldl`, since that was noted in the docs: https://github.com/NVIDIA/NVTX/#c-and-c

### CODE REVIEW

File: N/NVTX/build_tarballs.jl [hunk 1]
```diff
@@ -14,9 +14,10 @@ sources = [
 script = raw"""
 atomic_patch -p1 $WORKSPACE/srcdir/patches/nvtx-linux-limits.patch
 atomic_patch -p1 $WORKSPACE/srcdir/patches/nvtx-windows-case.patch
+atomic_patch -p1 $WORKSPACE/srcdir/patches/nvtx-pragma-hidden.patch
 cd ${WORKSPACE}/srcdir/
 mkdir -p ${libdir}
-${CC} -std=c99 -O2 -fPIC -shared -I${WORKSPACE}/srcdir/NVTX/c/include -o ${libdir}/libnvToolsExt.${dlext} nvtx.c 
+${CC} -std=c99 -O2 -fPIC -shared -ldl -I${WORKSPACE}/srcdir/NVTX/c/include -o ${libdir}/libnvToolsExt.${dlext} nvtx.c
```
Comment (giordano) [inline N/NVTX/build_tarballs.jl:20]:
```suggestion
if [[ "${target}" == *-linux-* ]]; then
    CFLAGS="-fPIC"
    LIBS="-ldl"
fi
${CC} -std=c99 -O2 "${CFLAGS}" -shared "${LIBS}" -I${WORKSPACE}/srcdir/NVTX/c/include -o ${libdir}/libnvToolsExt.${dlext} nvtx.c
```

Comment (simonbyrne) [reply N/NVTX/build_tarballs.jl]: Thanks!

**Comment by simonbyrne** [comment]
Looks like the upload and tag failed: https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=23837&view=logs&jobId=31684251-8f14-5149-f2b8-2d89411591c0&j=31684251-8f14-5149-f2b8-2d89411591c0&t=872aea0e-c9da-5abd-80d0-440e18b96cba

**Comment by giordano** [comment]
Something's very wrong: https://github.com/JuliaBinaryWrappers/NVTX_jll.jl/commit/394f4b777b2a74357426e3ce0f636e21c96dc41d.  The README.md correctly says version is v3.1.0+1, but the commit title is 3.1.0+0, which is wrong, and the upload is trying to overwrite the tag 3.1.0+0 .

**Comment by giordano** [comment]
I kicked the job again and that seemed to have worked.  But I don't understand why it got the wrong version number.  The version number used in the commit message and for the tag where to upload the tarballs comes from https://github.com/JuliaPackaging/Yggdrasil/blob/5bf229ace313d58ac9b54355eb0431028e36799f/.ci/register_package.jl#L32 and `BinaryBuilder.get_next_wrapper_version` triggers a registry update: https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/6d0e29382b7ed90de6d040335816a4f43259f823/src/AutoBuild.jl#L486-L487.  Perhaps the update failed and we didn't see it (the output is directed to `devnull`)?

**Comment by simonbyrne** [comment]
Thanks for your help!

**Pull request merged**